internal/backends/python/grab.go

```diff
@@ -0,0 +1,333 @@
+	"configparser",
+	"contextlib",
+	"contextvars",
+	"copy",
+	"copyreg",
+	"cProfile",
+	"crypt",
+	"csv",
+	"ctypes",
+	"curses",
+	"dataclasses",
+	"datetime",
+	"dbm",
+	"decimal",
+	"difflib",
+	"dis",
+	"doctest",
+	"email",
+	"encodings",
+	"ensurepip",
+	"enum",
+	"errno",
+	"faulthandler",
+	"fcntl",
+	"filecmp",
+	"fileinput",
+	"fnmatch",
+	"fractions",
+	"ftplib",
+	"functools",
+	"gc",
+	"getopt",
+	"getpass",
+	"gettext",
+	"glob",
+	"graphlib",
+	"grp",
+	"gzip",
+	"hashlib",
+	"heapq",
+	"hmac",
+	"html",
+	"http",
+	"idlelib",
+	"imaplib",
+	"imghdr",
+	"importlib",
+	"inspect",
+	"io",
+	"ipaddress",
+	"itertools",
+	"json",
+	"keyword",
+	"lib2to3",
+	"linecache",
+	"locale",
+	"logging",
+	"lzma",
+	"mailbox",
+	"mailcap",
+	"marshal",
+	"math",
+	"mimetypes",
+	"mmap",
+	"modulefinder",
+	"msilib",
+	"msicrt",
+	"multiprocessing",
+	"netrc",
+	"nis",
+	"nntplib",
+	"numbers",
+	"operator",
+	"optparse",
+	"os",
+	"ossaudiodev",
+	"parser",
+	"pathlib",
+	"pdb",
+	"pickle",
+	"pickletools",
+	"pipes",
+	"pkgutil",
+	"platform",
+	"plistlib",
+	"poplib",
+	"posix",
+	"pprint",
+	"profile",
+	"pstats",
+	"pty",
+	"pwd",
+	"py_compile",
+	"pyclbr",
+	"pydoc",
+	"queue",
+	"quopri",
+	"random",
+	"re",
+	"readline",
+	"reprlib",
+	"resource",
+	"rlcompleter",
+	"runpy",
+	"sched",
+	"secrets",
+	"select",
+	"selectors",
+	"shelve",
+	"shlex",
+	"shutil",
+	"signal",
+	"site",
+	"sitecustomize",
+	"smtplib",
+	"sndhdr",
+	"socket",
+	"socketserver",
+	"spwd",
+	"sqlite3",
+	"ssl",
+	"stat",
+	"statistics",
+	"string",
+	"stringprep",
+	"struct",
+	"subprocess",
+	"sunau",
+	"symtable",
+	"sys",
+	"sysconfig",
+	"syslog",
+	"tabnanny",
+	"tarfile",
+	"telnetlib",
+	"tempfile",
+	"termios",
+	"test",
+	"textwrap",
+	"threading",
+	"time",
+	"timeit",
+	"tkinter",
+	"token",
+	"tokenize",
+	"tomllib",
+	"trace",
+	"traceback",
+	"tracemalloc",
+	"tty",
+	"turtle",
+	"turtledemo",
+	"types",
+	"typing",
+	"unicodedata",
+	"unittest",
+	"urllib",
+	"usercustomize",
+	"uu",
+	"uuid",
+	"venv",
+	"warnings",
+	"wave",
+	"weakref",
+	"webbrowser",
+	"winreg",
+	"winsound",
+	"wsgiref",
+	"xdrlib",
+	"xml",
+	"xmlrpc",
+	"zipapp",
+	"zipfile",
+	"zipimport",
+	"zlib",
+	"zoneinfo",
+}
+
+func guess(python string) (map[api.PkgName]bool, bool) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		util.Die("couldn't get working directory: %s", err)
+	}
+
+	foundImportPaths, err := findImports(cwd)
+	if err != nil {
+		util.Die("couldn't guess imports: %s", err)
+	}
+
+	return filterImports(foundImportPaths)
+}
+
+func findImports(dir string) (map[string]bool, error) {
+	py := python.GetLanguage()
+	pkgs, err := util.GuessWithTreeSitter(dir, py, importsQuery, pyPathGlobs, pyIgnoreGlobs)
+
+	if err != nil {
+		return nil, err
+	}
+
+	foundImportPaths := map[string]bool{}
+	for _, pkg := range pkgs {
+		foundImportPaths[pkg] = true
+	}
+
+	return foundImportPaths, nil
+}
+
+func filterImports(foundPkgs map[string]bool) (map[api.PkgName]bool, bool) {
```

Comment:
Could you add a test for filtering imports as well?

Reply:
that doesn't work with the unit tests, since the `PYPI_MAP_DB` env var needs to be set. the filtering functionality is tested with the test-suite, though